src/Network/Cookie/Cookie.php

```diff
@@ -0,0 +1,256 @@
+<?php
+
+namespace Cake\Network\Cookie;
+
+use Cake\Core\InstanceConfigTrait;
+use Cake\I18n\Time;
+use Cake\Utility\Hash;
+use Cake\Utility\Security;
+use UnexpectedValueException;
+
+class Cookie
+{
+
+    const EXPIRES_FORMAT = 'U';
+    const TIMEOUT = '-1 month';
+
+    use InstanceConfigTrait;
+
+    /**
+     * Default config
+     *
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'path' => null,
+        'domain' => '',
+        'secure' => false,
+        'salt' => null,
+        'httpOnly' => false,
+        'encryption' => 'aes',
```

Comment:
In laravel framework they do encryption in middleware. So its an external process - cookies themselves are not aware of encryption and that may be the right approach.

So let's say we move cookies encryption to another class and handle it from there at some point of app runtime. There should be some way of telling this encrypter what to do with cookies -should they be encrypted and how as CookieComponent allows cookies to be configured that way on the name basis.